src/DependencyInjection/Compiler/ThumbnailCompilerPass.php

```diff
@@ -32,12 +33,23 @@ public function process(ContainerBuilder $container)
             'sonata.media.thumbnail.format'
         );
 
-        $reflectionClass = new \ReflectionClass($container->getParameterBag()->resolveValue($definition->getClass()));
+        $resolvedClass = $container->getParameterBag()->resolveValue($definition->getClass());
+        $reflectionClass = new \ReflectionClass($resolvedClass);
 
+        // NEXT_MAJOR: Check against the ResizableThumbnailInterface instead and remove the deprecation.
         if (!$reflectionClass->hasMethod('addResizer') || !$reflectionClass->getMethod('addResizer')->isPublic()) {
             return;
         }
 
+        if (!is_a($resolvedClass, ResizableThumbnailInterface::class, true)) {
+            @trigger_error(sprintf(
+                'Not implementing %s on a format thumbnail that uses resizers is deprecated since sonata-project/media-bundle 3.x'
```

Comment:
```suggestion
                'Not implementing %s on a thumbnail format that uses resizers is deprecated since sonata-project/media-bundle 3.x.'
```